### PR TITLE
Update GDASapp hash to move JCB into GDASapp

### PR DESCRIPTION
**Pull request by danholdaway**
# Description

Having JCB in global-workflow results in situations where the CI cannot satisfactorily test the code. This PR moves JCB into GDASapp.  The PR also bumps up the hash of GDASapp to what is in `feature/move_jcb`, which at time of writing is develop plus the absorption of JCB into GDASapp.

Once ready for merge I would suggest we merge the upstream changes into develop so a final hash of GDASapp can point to develop.

Note that I also took the changes from https://github.com/NOAA-EMC/global-workflow/pull/2641 to follow the testing @RussTreadon-NOAA has done.

# Type of change
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
Hera GDASapp Ctests

# Checklist
- [ ] Any dependent changes have been merged and published
- [X] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Review by CoryMartin-NOAA — APPROVED** [review]

File: sorc/gsi_utils.fd
Comment (aerorahul) [inline sorc/gsi_utils.fd:1]: Clicking on the link to the updated 4 files, it shows no change.
Are we sure this update is related and needed?

**Review by aerorahul — COMMENTED** [review]
looks good with the exception of gsi_utils submodule.

**Comment by RussTreadon-NOAA** [comment]
g-w `develop` currently points at [gsi_utils.fd @ d940406](https://github.com/NOAA-EMC/GSI-Utils/tree/d9404061611553459394173c3ff33116db306326).   We should stick with this `GSI-utils` hash or jump to the current head of `GSI-utils`, [4332814](https://github.com/NOAA-EMC/GSI-utils/commit/4332814529465ab8eb58e43a38227b952ebfca49).  

[gsi_utils.fd @ bb03e17](https://github.com/NOAA-EMC/GSI-Utils/tree/bb03e172e0d0d9c56d6da7788ca033bfb5ef5119) is two commits behind  [gsi_utils.fd @ d940406](https://github.com/NOAA-EMC/GSI-Utils/tree/d9404061611553459394173c3ff33116db306326)

File: sorc/gsi_utils.fd
Comment (danholdaway) [reply sorc/gsi_utils.fd:1]: Thanks @RussTreadon-NOAA @aerorahul. I was following the change in Russ's PR and assuming there was something new we needed from GSI utils now. If now lets just keep to what's in develop.

**Review by aerorahul — APPROVED** [review]
Looks good to me.

**Comment by RussTreadon-NOAA** [comment]
**Orion test**

Install `danholdaway:feature/move_jcb` at 4d3150ca on Orion.  Build apps and run GDASApp ctests.  All tests pass except one
```
98% tests passed, 1 tests failed out of 47

Label Time Summary:
gdas-utils    =  15.66 sec*proc (9 tests)
script        =  15.66 sec*proc (9 tests)

Total Test time (real) = 4102.87 sec

The following tests FAILED:
        1849 - test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_VRFY (Failed)
```
This failure also occurs when using GDASApp `develop` and has been reported via GDASApp issue [#1153](https://github.com/NOAA-EMC/GDASApp/issues/1153).  GDASApp PR [#1157](https://github.com/NOAA-EMC/GDASApp/pull/1157) removed, for the time being, this test from the list of active GDASApp ctests.

Enable C96C48_ufs_hybatmDA CI on Orion.  Run CI for this set up.  All jobs successfully run to completion.  All DA jobs successfully ran to completion.  Only the gfsfcst and downstream jobs remain to be run.

**Review by RussTreadon-NOAA — APPROVED** [review]
Changes look good.  test_gdasapp and JEDI ATM CI tests yield acceptable results.

Approve.

**Comment by RussTreadon-NOAA** [comment]
Just closing the loop on Orion CI tests.  All jobs for C96C48_ufs_hybatmDA CI are now complete and successful
```
Orion-login-2:/work2/noaa/stmp/rtreadon/EXPDIR/prmove$ rocotostat -d prmove.db -w prmove.xml -c all -s
   CYCLE         STATE           ACTIVATED              DEACTIVATED
202402231800        Done    Jun 07 2024 13:56:28    Jun 07 2024 14:20:39
202402240000        Done    Jun 07 2024 13:56:28    Jun 07 2024 19:24:25
```

**Comment by WalterKolczynski-NOAA** [comment]
@danholdaway The gdas conflict after Russ's PR got merged needs to be resolved.

**Comment by danholdaway** [comment]
@WalterKolczynski-NOAA all up to date.

**Comment by WalterKolczynski-NOAA** [comment]
@danholdaway thanks

**Comment by DavidHuber-NOAA** [comment]
Manually checking, all tests passed on Hera.  I think the Java controller crashed at some point.

**Comment by TerrenceMcGuinness-NOAA** [comment]
@DavidHuber-NOAA Yes I noticed one view had all of the experiments running in the pipeline but the controller had given up because it came up and down.  I was just confirming they all passed too.

**Confirmation:**

```
Terry.McGuinness (hfe09) utils (develop) $ ~/bin/check_expdir.sh /scratch1/NCEPDEV/global/CI/2665/RUNTESTS/EXPDIR
C48_ATM_4de31999
DONE
C48_S2SWA_gefs_4de31999
DONE
C48_S2SW_4de31999
DONE
C48mx500_3DVarAOWCDA_4de31999
DONE
C96C48_hybatmDA_4de31999
DONE
C96_atm3DVar_4de31999
DONE
C96_atmaerosnowDA_4de31999
DONE
```

**Pull request merged**